extensions/amp-a4a/amp-a4a-format.md

```diff
@@ -0,0 +1,358 @@
+```
+
+Note that the same rules about mutations to the boilerplate text apply.
+
+### CSS
+
+1. `position:fixed` and `position:sticky` are prohibited in 
+creative CSS.
+
+  _Rationale_: position:fixed breaks out of shadow DOM, which A4A depends on.  Also, Ads in AMP are already not allowed to use fixed position.
+
+1. Creative CSS is limited to 20kb.
+
+  _Rationale_: Large CSS blocks bloat the creative, increase network latency, and degrade page performance.
+
+1. CSS: transition and animation are subject to additional restrictions.
+
+  _Rationale_: AMP must be able to control all animations belonging to an ad, so that it can stop them when the ad is not on screen or system resources are very low.
+
+#### CSS Animations and Transitions
+
+##### Selectors
+
+The `transition` and `animation` properties are only allowed on selectors that:
+- Contain only `transition`, `animation`, `transform`, `visibility`, or 
+  `opacity` properties.
+- Start with `.amp-animate` followed by a space.
+
+  _Rationale:_ This allows the AMP runtime to remove this  class from context
+   to deactivate animations, when necessary for page performance.
+
+**Good**
+```css
+.amp-animate .box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+**Bad**
+
+Property not allowed in CSS class.
+```css
+.amp-animate .box {
+    color: red;  // non-animation property not allowed in animation selector
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+Missing context class `.amp-animate`.
+```css
+.box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+##### Transitionable and animatable properties
+
+The only properties that may be transitioned are opacity and transform.
+([Rationale](http://www.html5rocks.com/en/tutorials/speed/high-performance-animations/))
+
+**Good**
+```css
+transition: transform 2s;
+```
+
+**Bad**
+```css
+transition: background-color 2s;
+```
+
+**Good**
+```css
+@keyframes turn {
+  from {
+    transform: rotate(180deg);
+  }
+  
+  to {
+    transform: rotate(90deg);
+  }
+}
+```
+
+**Bad**
+```css
+@keyframes slidein {
+  from {
+    margin-left:100%;
+    width:300%
+  }
+  
+  to {
+    margin-left:0%;
+    width:100%;
+  }
+}
+```
+
+
+### AMP Extensions
+
+The following are _allowed_ AMP extension modules in an A4A creative. Extensions
+not explicitly allowed are prohibited.
+
+Most of the omissions are either for performance or to make A4A creatives 
+simpler to analyze.
+
```

Comment:
Good point.  In part, I was just following form, since AMP spec itself talks about, e.g., tags with a whitelist.  More broadly, though, in the hopes that other groups will implement validators and signing services, I'd like to encourage them to think in terms of whitelists as well.